test/test_utils.py

```diff
@@ -249,17 +249,36 @@ def _test_sanitize_path(self):
         self.assertEqual(sanitize_path('abc/def...'), 'abc\\def..#')
         self.assertEqual(sanitize_path('abc.../def'), 'abc..#\\def')
         self.assertEqual(sanitize_path('abc.../def...'), 'abc..#\\def..#')
-
-        self.assertEqual(sanitize_path('../abc'), '..\\abc')
-        self.assertEqual(sanitize_path('../../abc'), '..\\..\\abc')
-        self.assertEqual(sanitize_path('./abc'), 'abc')
-        self.assertEqual(sanitize_path('./../abc'), '..\\abc')
-
-        self.assertEqual(sanitize_path('\\abc'), '\\abc')
-        self.assertEqual(sanitize_path('C:abc'), 'C:abc')
-        self.assertEqual(sanitize_path('C:abc\\..\\'), 'C:..')
         self.assertEqual(sanitize_path('C:\\abc:%(title)s.%(ext)s'), 'C:\\abc#%(title)s.%(ext)s')
 
+        # Check with nt._path_normpath if available
+        try:
+            import nt
+
+            nt_path_normpath = getattr(nt, '_path_normpath', None)
+        except Exception:
+            nt_path_normpath = None
```

Comment:
In the [CPython source](https://github.com/python/cpython/blob/3.13/Lib/ntpath.py#LC510), a simpler tactic like this (what I would have expected) is used:
```suggestion
        try:
            from nt import _path_normpath as nt_path_normpath
        except ImportError:
            nt_path_normpath = None
```
But then if the import succeeds here it would have succeeded in the Python library. This means having a sufficiently new Python (>=3.11) running on Windows. In that case `os.path.normpath` is an alias of `nt_path_normpath`: shouldn't `sanitize_path()` just punt to `os.path.normpath()` then? And then you wonder whether the library fallback code for 3.9 and 3.10 could replace the custom code in `sanitize_path()`.

Reply:
The mock on `sys.platform` is not used by `os.path.normpath()`, so it doesn't affect the path processing on POSIX. To make the test more useful, since `sanitize_path()` is trivial when not run on Windows, a mock setting `os.path` to `ntpath` should be added.

Then in 2.7, 3.5, 3.9, I find that the result of `os.path.normpath()` against the list of `test, expected` is the `expected` just as with sanitize_path().

Reply:
> In that case `os.path.normpath` is an alias of `nt_path_normpath`

The functionality was *backported* to 3.11, and `sanitize_path` does not only normalize.

> And then you wonder whether the library fallback code for 3.9 and 3.10 could replace the custom code

No, see https://github.com/yt-dlp/yt-dlp/issues/8137

> The mock on `sys.platform` is not used by `os.path.normpath()`

it is used by `sanitize_path` not `os.path.normpath`. We use it to check windows handling explicitly.

> But these cases in the preceding tests give different results

The tests are split into two parts: one with replacements, one without. Replacements are part of the sanitization to produce valid filenames and therefore cannot be checked against windows api.

Reply:
Essentially I'm proposing s/t like this upstream:
```py
    import ntpath
    @unittest.mock('sys.platform', 'win32')
    @unittest.mock('os.path', ntpath)
    def test_sanitize_path(self):
        ...
```
Then the test routine runs all the tests non-trivially and successfully on Linux.

Depending on how well the _ntpath_ module emulates Windows on other platforms this makes the test useful on POSIX-ish platforms too. But #8137 (thx) suggests that the functionality on Windows might actually vary by Python and OS version, even while the POSIX fallback is consistent. I guess I'll find out.